### PR TITLE
PVP-199

### PR DESCRIPTION
**Pull request by KhajiitLikeToSneak**
- Applied design that was created in PVP-159
- Added functionality to edit task details

Distance picker always starts from 0. Couldn't manage to make it start from the current task distance.

_Dear reviewers sorry for a mess_ 
_Will do a cleanup and make everything more abstract on the weekend_

### CODE REVIEW

**Review by vytjar — APPROVED** [review]

File: app/src/main/java/com/pvp/app/ui/common/Components.kt [hunk 1]
```diff
@@ -375,7 +375,7 @@ fun EditableInfoItem(
     label: String,
     onConfirm: (String) -> Unit,
     onDismiss: () -> Unit,
-    value: String
+    value: String,
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/common/Components.kt:378]: Redundant

File: app/src/main/java/com/pvp/app/ui/common/Pickers.kt [hunk 2]
```diff
@@ -78,7 +80,10 @@ fun <T> Picker(
         snapshotFlow { stateList.firstVisibleItemIndex }
             .map { index -> getItem(index + itemsInMiddle) }
             .distinctUntilChanged()
-            .collect { item -> state.value = item }
+            .collect { item ->
+                state.value = item
+                onChange(item)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/common/Pickers.kt:85]: Separate

File: app/src/main/java/com/pvp/app/ui/common/Pickers.kt [hunk 3]
```diff
@@ -117,4 +122,36 @@ fun <T> Picker(
...
+            modifier = Modifier.weight(1f),
+            startIndex = selectedHour.value,
+            onChange = { onChange(it, selectedMinute.value) }
+        )
+
+        Text(text = ":", modifier = Modifier.align(Alignment.CenterVertically))
+
+        Picker(
+            items = (0..59).toList(),
+            label = { it.toString().padStart(2, '0') },
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/common/Pickers.kt:149]: Split

File: app/src/main/java/com/pvp/app/ui/common/Pickers.kt [hunk 4]
```diff
@@ -117,4 +122,36 @@ fun <T> Picker(
...
+) {
+    Row(modifier = modifier) {
+        Picker(
+            items = (0..23).toList(),
+            label = { "$it" },
+            state = selectedHour,
+            visibleItemsCount = 3,
+            modifier = Modifier.weight(1f),
+            startIndex = selectedHour.value,
+            onChange = { onChange(it, selectedMinute.value) }
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/common/Pickers.kt:142]: Split

File: app/src/main/java/com/pvp/app/ui/common/Pickers.kt [hunk 5]
```diff
@@ -117,4 +122,36 @@ fun <T> Picker(
...
+        )
+
+        Text(text = ":", modifier = Modifier.align(Alignment.CenterVertically))
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/common/Pickers.kt:145]: Split

File: app/src/main/java/com/pvp/app/ui/common/Pickers.kt [hunk 6]
```diff
@@ -117,4 +122,36 @@ fun <T> Picker(
...
+
+        Picker(
+            items = (0..59).toList(),
+            label = { it.toString().padStart(2, '0') },
+            state = selectedMinute,
+            visibleItemsCount = 3,
+            modifier = Modifier.weight(1f),
+            startIndex = selectedMinute.value,
+            onChange = { onChange(selectedHour.value, it) }
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/common/Pickers.kt:154]: Split

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 7]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+    onDismissRequest: () -> Unit,
+    showDialog: Boolean,
+    model: TaskViewModel = hiltViewModel()
+) {
+    var title by remember { mutableStateOf(task.title) }
+    var description by remember { mutableStateOf(task.description) }
+    var duration by remember { mutableStateOf(task.duration) }
+    var scheduledAt by remember { mutableStateOf(task.scheduledAt) }
+
+
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:693]: Redundant line

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 8]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+    if (showDialog) {
+        Dialog(onDismissRequest = onDismissRequest) {
+            Surface(
+                shape = RoundedCornerShape(10.dp),
+                color = MaterialTheme.colorScheme.surfaceContainer,
+            ) {
+                Column(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding( end = 15.dp, start = 15.dp, top = 15.dp )
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:703]: Split

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 9]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                            Column {
+                                Text(
+                                    text = "Duration: ${duration?.toMinutes()} minutes",
+                                    style = TextStyle(fontSize = 16.sp),
+                                    modifier = Modifier
+                                        .fillMaxWidth()
+                                        .padding(start = 4.dp)
+                                )
+
+                                Slider(
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:754]: Slider is a pain in the *** when selecting specific duration time.

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 10]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                                    value = duration?.toMinutes()?.toFloat() ?: 0f,
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:755]: Split

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 11]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                        dialogTitle = { Text("Editing duration") },
+                        label = "Duration",
+                        onConfirm = { duration = duration },
+                        onDismiss = { duration = task.duration },
+                        value = "${duration?.toMinutes()} minutes"
+                    )
+
+                    val selectedHour = rememberPickerState(scheduledAt.hour)
+                    val selectedMinute = rememberPickerState(scheduledAt.minute)
+                    EditableInfoItem(
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:780]: Separate

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 12]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                        dialogContent = {
+                            TimePicker(
+                                selectedHour = selectedHour,
+                                selectedMinute = selectedMinute,
+                                onChange = { hour, minute ->
+                                    scheduledAt = scheduledAt.withHour(hour).withMinute(minute)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:786]: Split

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 13]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                                }
+                            )
+                        },
+                        dialogTitle = { Text("Editing scheduled at") },
+                        label = "Scheduled at",
+                        onConfirm = {
+                            scheduledAt = scheduledAt.withHour(selectedHour.value)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:793]: Split

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 14]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                                .withMinute(selectedMinute.value)
+                        },
+                        onDismiss = { scheduledAt = task.scheduledAt },
+                        value = "${scheduledAt.format(DateTimeFormatter.ofPattern("hh:mm a"))} - " +
+                                "${(scheduledAt.plus(duration)).format(DateTimeFormatter.ofPattern("hh:mm a"))}"
+                    )
+
+
+
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:802]: Redundant lines (2)

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 15]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                        },
+                        label = "Date",
+                        onConfirm = { },
+                        value = "${scheduledAt.format(DateTimeFormatter.ofPattern("yyyy-MM-dd, EEEE"))}"
+                    )
+
+                    Row(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(horizontal = 24.dp, vertical = 15.dp),
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:824]: Split

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 16]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                        var showConfirmationDialog by remember { mutableStateOf(false) }
+
+                        OutlinedButton(
+                            onClick = { showConfirmationDialog = true },
+                            shape = MaterialTheme.shapes.extraLarge,
+                            border = BorderStroke(1.dp, Color.Red),
+                            colors = ButtonDefaults.outlinedButtonColors(contentColor = Color.Red)
+                        ) {
+                            Row(verticalAlignment = Alignment.CenterVertically) {
+                                Icon(Icons.Outlined.Delete, contentDescription = "Remove task")
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:836]: Split

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 17]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                                Text("Delete", style = MaterialTheme.typography.titleSmall)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:837]: Split, separate

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 18]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                        }
+
+                        if (showConfirmationDialog) {
+                            AlertDialog(
+                                onDismissRequest = { showConfirmationDialog = false },
+                                confirmButton = {
+                                    Button(
+                                        onClick = {
+                                            model.remove(task)
+                                            showConfirmationDialog = false
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:848]: Separate

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 19]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+
+                        Button(
+                            onClick = {
+                                model.update({ task ->
+                                    task.title = title
+                                    task.description = description
+                                    task.duration = duration
+                                    task.scheduledAt = scheduledAt
+                                }, task)
+                                onDismissRequest()
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:875]: Split, separate

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 20]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+            .padding(8.dp)
+            .fillMaxWidth()
+    ) {
+        Column(
+            modifier = Modifier.fillMaxWidth(),
+            horizontalAlignment = Alignment.Start,
+            verticalArrangement = Arrangement.Center
+        ) {
+            Row(
+                modifier = Modifier.fillMaxWidth().padding(end = 3.dp),
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:911]: Split

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 21]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                    modifier = Modifier
+                        .size(24.dp)
+                        .clickable { showDialog = true }
+                )
+            }
+
+            Text(text = value)
+        }
+
+        dialogContent(showDialog, { showDialog = false }, { selectedDate ->
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:931]: Split

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 22]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                        dialogTitle = { Text("Editing title") },
+                        label = "Title",
+                        onConfirm = { title = it },
+                        onDismiss = { title = task.title },
+                        value = title
+                    )
+
+                    EditableInfoItem(
+                        dialogContent = {
+                            var isExpanded by remember { mutableStateOf(false) }
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:986]: Separate

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 23]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                        },
+                        dialogTitle = { Text("Editing description") },
+                        label = "Description",
+                        onConfirm = { description = it },
+                        onDismiss = { description = task.description },
+                        value = description ?: ""
+                    )
+
+                    if (activity!!.supportsDistanceMetrics) {
+                        val statePickerDistance = rememberPickerState(0.0)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:1045]: Separate

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 24]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                            onConfirm = { duration = duration },
+                            onDismiss = { duration = task.duration },
+                            value = "${duration?.toMinutes()} minutes"
+                        )
+                    }
+
+                    if (activity!!.supportsDistanceMetrics) {
+                        val selectedHour = rememberPickerState(scheduledAt.hour)
+                        val selectedMinute = rememberPickerState(scheduledAt.minute)
+                        EditableInfoItem(
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:1107]: Separate

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 25]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                            dialogContent = {
+                                TimePicker(
+                                    selectedHour = selectedHour,
+                                    selectedMinute = selectedMinute,
+                                    onChange = { hour, minute ->
+                                        scheduledAt = scheduledAt.withHour(hour).withMinute(minute)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:1113]: Split

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 26]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                                    }
+                                )
+                            },
+                            dialogTitle = { Text("Editing scheduled at") },
+                            label = "Scheduled at",
+                            onConfirm = {
+                                scheduledAt = scheduledAt.withHour(selectedHour.value)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:1120]: Split

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 27]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                    } else {
+                        val selectedHour = rememberPickerState(scheduledAt.hour)
+                        val selectedMinute = rememberPickerState(scheduledAt.minute)
+                        EditableInfoItem(
+                            dialogContent = {
+                                TimePicker(
+                                    selectedHour = selectedHour,
+                                    selectedMinute = selectedMinute,
+                                    onChange = { hour, minute ->
+                                        scheduledAt = scheduledAt.withHour(hour).withMinute(minute)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:1135]: Split

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 28]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                                scheduledAt = scheduledAt.withHour(selectedHour.value)
+                                    .withMinute(selectedMinute.value)
+                            },
+                            onDismiss = { scheduledAt = task.scheduledAt },
+                            value = "${scheduledAt.format(DateTimeFormatter.ofPattern("hh:mm a"))}"
+                        )
+                    } else {
+                        val selectedHour = rememberPickerState(scheduledAt.hour)
+                        val selectedMinute = rememberPickerState(scheduledAt.minute)
+                        EditableInfoItem(
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:1129]: Separate

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 29]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                                    selectedMinute = selectedMinute,
+                                    onChange = { hour, minute ->
+                                        scheduledAt = scheduledAt.withHour(hour).withMinute(minute)
+                                    }
+                                )
+                            },
+                            dialogTitle = { Text("Editing scheduled at") },
+                            label = "Scheduled at",
+                            onConfirm = {
+                                scheduledAt = scheduledAt.withHour(selectedHour.value)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:1142]: Split

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 30]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                        },
+                        label = "Date",
+                        onConfirm = { },
+                        value = "${scheduledAt.format(DateTimeFormatter.ofPattern("yyyy-MM-dd, EEEE"))}"
+                    )
+
+                    Row(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(horizontal = 24.dp, vertical = 15.dp),
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:1172]: Split

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 31]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+
+                        OutlinedButton(
+                            onClick = { showConfirmationDialog = true },
+                            shape = MaterialTheme.shapes.extraLarge,
+                            border = BorderStroke(1.dp, Color.Red),
+                            colors = ButtonDefaults.outlinedButtonColors(contentColor = Color.Red)
+                        ) {
+                            Row(verticalAlignment = Alignment.CenterVertically) {
+                                Icon(Icons.Outlined.Delete, contentDescription = "Remove task")
+                                Text("Delete", style = MaterialTheme.typography.titleSmall)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:1185]: Split, separate both

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 32]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                        }
+
+                        if (showConfirmationDialog) {
+                            AlertDialog(
+                                onDismissRequest = { showConfirmationDialog = false },
+                                confirmButton = {
+                                    Button(
+                                        onClick = {
+                                            model.remove(task)
+                                            showConfirmationDialog = false
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:1196]: Separate

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 33]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                                        }
+                                    }
+                                    distance?.let {
+                                        if (task is SportTask) {
+                                            task.distance = it
+                                        }
+                                    }
+                                    task.duration = duration
+                                    task.scheduledAt = scheduledAt
+                                }, task)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:1231]: Split, separate

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 34]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+    if (showDialog) {
+        Dialog(onDismissRequest = onDismissRequest) {
+            Surface(
+                shape = RoundedCornerShape(10.dp),
+                color = MaterialTheme.colorScheme.surfaceContainer,
+            ) {
+                Column(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(end = 15.dp, start = 15.dp, top = 15.dp)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:1266]: Split

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 35]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                        dialogTitle = { Text("Editing duration") },
+                        label = "Duration",
+                        onConfirm = { duration = duration },
+                        onDismiss = { duration = task.duration },
+                        value = "${duration?.toMinutes()} minutes"
+                    )
+
+                    val selectedHour = rememberPickerState(scheduledAt.hour)
+                    val selectedMinute = rememberPickerState(scheduledAt.minute)
+                    EditableInfoItem(
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:1361]: Separate

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 36]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                        dialogContent = {
+                            TimePicker(
+                                selectedHour = selectedHour,
+                                selectedMinute = selectedMinute,
+                                onChange = { hour, minute ->
+                                    scheduledAt = scheduledAt.withHour(hour).withMinute(minute)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:1367]: Split

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 37]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                                }
+                            )
+                        },
+                        dialogTitle = { Text("Editing scheduled at") },
+                        label = "Scheduled at",
+                        onConfirm = {
+                            scheduledAt = scheduledAt.withHour(selectedHour.value)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:1374]: Split

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 38]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+
+                        OutlinedButton(
+                            onClick = { showConfirmationDialog = true },
+                            shape = MaterialTheme.shapes.extraLarge,
+                            border = BorderStroke(1.dp, Color.Red),
+                            colors = ButtonDefaults.outlinedButtonColors(contentColor = Color.Red)
+                        ) {
+                            Row(verticalAlignment = Alignment.CenterVertically) {
+                                Icon(Icons.Outlined.Delete, contentDescription = "Remove task")
+                                Text("Delete", style = MaterialTheme.typography.titleSmall)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:1417]: Split, separate both

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 39]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                            .fillMaxWidth()
+                            .padding(horizontal = 24.dp, vertical = 15.dp),
+                        horizontalArrangement = Arrangement.SpaceBetween
+                    ) {
+                        var showConfirmationDialog by remember { mutableStateOf(false) }
+
+                        OutlinedButton(
+                            onClick = { showConfirmationDialog = true },
+                            shape = MaterialTheme.shapes.extraLarge,
+                            border = BorderStroke(1.dp, Color.Red),
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:1412]: Split

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 40]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                        }
+
+                        if (showConfirmationDialog) {
+                            AlertDialog(
+                                onDismissRequest = { showConfirmationDialog = false },
+                                confirmButton = {
+                                    Button(
+                                        onClick = {
+                                            model.remove(task)
+                                            showConfirmationDialog = false
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:1428]: Split

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 41]
```diff
@@ -635,6 +678,792 @@ fun TaskBox(
...
+                        Button(
+                            onClick = {
+                                model.update({ task ->
+                                    task.title = title
+                                    task.description = description
+                                    task.recipe = recipe
+                                    task.duration = duration
+                                    task.scheduledAt = scheduledAt
+                                }, task)
+                                onDismissRequest()
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:1455]: Split, separate

**Review by klaidoshka — APPROVED** [review]

**Pull request merged**